src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -76,26 +94,38 @@ public final void publishWithExpiry(final Message message, final long expiryInMs
                     .headers(ImmutableMap.of(MESSAGE_EXPIRY_TEXT, expiresAt))
                     .build();
         }
-        publish(message, properties);
+        publish(message, properties, RMQOperations.PUBLISH_WITH_EXPIRY.name());
     }
 
     public final void publish(final Message message) throws Exception {
-        publish(message, MessageProperties.MINIMAL_PERSISTENT_BASIC);
+        publish(message, MessageProperties.MINIMAL_PERSISTENT_BASIC, RMQOperations.PUBLISH.name());
     }
 
-    public final void publish(final Message message, final AMQP.BasicProperties properties) throws Exception {
+    public final void publish(final Message message, final AMQP.BasicProperties properties, final String operation) throws Exception {
         String routingKey;
         if (config.isSharded()) {
             routingKey = NamingUtils.getShardedQueueName(queueName, getShardId());
         } else {
             routingKey = queueName;
         }
-        val enrichedProperties = getEnrichedProperties(properties);
-        publishChannel.basicPublish(config.getExchange(), routingKey, enrichedProperties, mapper().writeValueAsBytes(message));
+        val context = ObserverContext.builder().operation(operation).queueName(routingKey).build();
```

Comment:
sending shardedQueueName might be an issue, all the metrics will get changed if sharding is enabled. Revisit this